azure-mgmt-keyvault/src/main/java/com/microsoft/azure/management/keyvault/implementation/VaultsImpl.java

```diff
@@ -97,4 +101,42 @@ protected VaultImpl wrapModel(VaultInner vaultInner) {
     public Observable<Vault> listByResourceGroupAsync(String resourceGroupName) {
         return wrapPageAsync(this.inner().listByResourceGroupAsync(resourceGroupName));
     }
+	
+	@Override
+	public PagedList<DeletedVault> listDeleted() {
+		return wrapPage(this.inner().listDeleted());
+	}
+
+	//Hacky method to convert from DeletedVaultInner to DeletedVault
+	private PagedList<DeletedVault> wrapPage(PagedList<DeletedVaultInner> listDeleted) {
+		PagedListConverter<DeletedVaultInner, DeletedVault> converter =  
+				new PagedListConverter<DeletedVaultInner, DeletedVault> () {
+					@Override
+					public Observable<DeletedVault> typeConvertAsync(DeletedVaultInner inner) {
+						return Observable.just((DeletedVault) new DeletedVaultImpl(inner));
+					}
+		};
+		return converter.convert(listDeleted);
+	}
```

Comment:
This is correct.

Just make sure we are not using the name `wrapPage`, we could move converter variable def to `public PagedList<DeletedVault> listDeleted()` method.